pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1459,6 +1460,30 @@ private ByteBuf processMessageChunk(ByteBuf compressedPayload, MessageMetadata m
         // discard message if chunk is out-of-order
         if (chunkedMsgCtx == null || chunkedMsgCtx.chunkedMsgBuffer == null
                 || msgMetadata.getChunkId() != (chunkedMsgCtx.lastChunkedMessageId + 1)) {
+            // Filter and ack duplicated chunks instead of discard ctx.
+            // For example:
+            //     Chunk-1 sequence ID: 0, chunk ID: 0, msgID: 1:1
+            //     Chunk-2 sequence ID: 0, chunk ID: 1, msgID: 1:2
+            //     Chunk-3 sequence ID: 0, chunk ID: 2, msgID: 1:3
+            //     Chunk-4 sequence ID: 0, chunk ID: 1, msgID: 1:4
+            //     Chunk-5 sequence ID: 0, chunk ID: 2, msgID: 1:5
+            //     Chunk-6 sequence ID: 0, chunk ID: 3, msgID: 1:6
+            // We should filter and ack chunk-4 and chunk-5.
+            if (chunkedMsgCtx != null && msgMetadata.getChunkId() <= chunkedMsgCtx.lastChunkedMessageId) {
```

Comment:
What's the expected behavior from the old version of clients that do not have this logic?
The old clients will not be able to handle any partial or duplicate chunk messages and will get stuck without this logic. 
It appeaars that chunking dedup will only work with this change on both client and brokers.

Reply:
What do you mean?
The behavior of the old client should be as follows:
```markdown
                 Chunk-1 sequence ID: 0, chunk ID: 0, msgID: 1:1
                 Chunk-2 sequence ID: 0, chunk ID: 1, msgID: 1:2
                 Chunk-3 sequence ID: 0, chunk ID: 2, msgID: 1:3
                 Chunk-4 sequence ID: 0, chunk ID: 1, msgID: 1:4 // out of order, release chunk-4 and discard ctx
                 Chunk-5 sequence ID: 0, chunk ID: 2, msgID: 1:5
                 Chunk-6 sequence ID: 0, chunk ID: 3, msgID: 1:6
```